compiler/src/org.graalvm.compiler.replacements/src/org/graalvm/compiler/replacements/StringLatin1Substitutions.java

```diff
@@ -78,20 +78,12 @@ public static int indexOf(byte[] value, int ch, int origFromIndex) {
     }
 
     @MethodSubstitution
-    public static int indexOf(byte[] source, int sourceCount, byte[] target, int targetCount, int origFromIndex) {
-        int fromIndex = origFromIndex;
-        if (injectBranchProbability(UNLIKELY_PROBABILITY, fromIndex >= sourceCount)) {
-            return (targetCount == 0 ? sourceCount : -1);
-        }
-        if (injectBranchProbability(UNLIKELY_PROBABILITY, fromIndex < 0)) {
-            fromIndex = 0;
-        }
-        if (injectBranchProbability(UNLIKELY_PROBABILITY, targetCount == 0)) {
-            // The empty string is in every string.
-            return fromIndex;
-        }
+    public static int indexOf(byte[] source, int sourceCount, byte[] target, int targetCount, int fromIndex) {
+        ReplacementsUtil.dynamicAssert(fromIndex >= 0, "StringLatin1.indexOf invalid args: fromIndex negative");
+        ReplacementsUtil.dynamicAssert(fromIndex < sourceCount, "StringLatin1.indexOf invalid args: fromIndex >= length(target)");
```

Comment:
This assumption isn't valid.

Reply:
You're right. If StringLatin1.indexOf is called from somewhere other than String.indexOf method, i.e., without going through the checks put by the indexOf method, then this assumption would be invalid.